Rover/Map.cpp

```diff
@@ -0,0 +1,327 @@
+	   q.second <= std::max(p.second, r.second) && q.second >= std::min(p.second, r.second));
+}
+
+bool RoverPathfinding::Map::segments_intersect(point p1, point p2, point q1, point q2)
+{
+    int o1 = orientation(p1, q1, p2);
+    int o2 = orientation(p1, q1, q2);
+    int o3 = orientation(p2, q2, p1);
+    int o4 = orientation(p2, q2, q1);
+ 
+    if (o1 != o2 && o3 != o4)
+        return(true);
+ 
+    if (o1 == 0 && on_segment(p1, p2, q1))
+	return(true);
+    
+    if (o2 == 0 && on_segment(p1, q2, q1))
+	return(true);
+ 
+    if (o3 == 0 && on_segment(p2, p1, q2))
+	return(true);
+ 
+    if (o4 == 0 && on_segment(p2, q1, q2))
+	return(true);
+ 
+    return(false);
+}
+
+std::pair<RoverPathfinding::point, RoverPathfinding::point> RoverPathfinding::Map::add_length_to_line_segment(point p, point q, float length)
+{
+    point pq = std::make_pair(q.first - p.first, q.second - p.second); //vector
+    float len = sqrt(pq.first * pq.first + pq.second * pq.second);
+    pq.first = length * pq.first / len;
+    pq.second = length * pq.second / len;
+
+    point p1 = std::make_pair(q.first + pq.first, q.second + pq.second);
+    point p2 = std::make_pair(p.first - pq.first, p.second - pq.second);
+    return(std::make_pair(p1, p2));   
+}
+
+RoverPathfinding::point RoverPathfinding::Map::center_point_with_radius(RoverPathfinding::point cur, RoverPathfinding::point p, RoverPathfinding::point q, float R)
+{
+    point vec = std::make_pair(-p.second + q.second, p.first - q.first);
+    float len = sqrt((vec.first * vec.first) + (vec.second * vec.second));
+    vec.first = vec.first * R / len;
+    vec.second = vec.second * R / len;
+    point result = std::make_pair((p.first + q.first) / 2.0f, (p.second + q.second) / 2.0f);
+    int o = orientation(cur, p, q);
+    if(o == 1)
+    {
+	result.first += vec.first;
+	result.second += vec.second;
+    }
+    else //o is 2
+    {
+	result.first -= vec.first;
+	result.second -= vec.second;
+    }
+    return(result);
+}
+
+float RoverPathfinding::Map::dist_sq(point p1, point p2)
+{
+    return((p1.first - p2.first) * (p1.first - p2.first) + (p1.second - p2.second) * (p1.second - p2.second));
+}
+
+bool RoverPathfinding::Map::within_radius(point p1, point p2, float R)
+{
+    return(dist_sq(p1, p2) <= R * R);
+}
+
+void RoverPathfinding::Map::add_edge(int n1, int n2)
+{
+    float dist = sqrt(dist_sq(nodes[n1].coord, nodes[n2].coord));
+    
+    auto n1_to_n2 = std::make_pair(n2, dist);
+    nodes[n1].connection.push_back(n1_to_n2);
+
+    auto n2_to_n1 = std::make_pair(n1, dist);
+    nodes[n2].connection.push_back(n2_to_n1);   
+}
+
+int RoverPathfinding::Map::create_node(point coord)
+{
+    node n;
+    n.prev = -1;
+    n.dist_to = INFINITY;
+    n.coord = coord;
+    nodes.push_back(n);
+    return(nodes.size() - 1);
+}
+
+std::vector<RoverPathfinding::node> RoverPathfinding::Map::build_graph(point cur, point tar)
+{
+    std::queue<int> q;
```

Comment:
Rename this to something which describes it's purpose and mentions what the int values inside mean (like unprocessedNodeIndexes)